src/restic/tag_list.go

```diff
@@ -0,0 +1,62 @@
+package restic
+
+import (
+	"fmt"
+	"strings"
+)
+
+// TagList is a list of tags.
+type TagList []string
+
+// SplitTagList splits a string into a list of tags. The tags in the string
+// need to be separated by commas. Whitespace is stripped around the individual
+// tags.
+func SplitTagList(s string) (l TagList) {
+	for _, t := range strings.Split(s, ",") {
+		l = append(l, strings.TrimSpace(t))
+	}
+	return l
+}
+
+func (l TagList) String() string {
+	return "[" + strings.Join(l, ", ") + "]"
+}
+
+// Set updates the TagList's value.
+func (l *TagList) Set(s string) error {
+	*l = SplitTagList(s)
+	return nil
+}
+
+// Type returns a description of the type.
+func (TagList) Type() string {
+	return "TagList"
+}
+
+// TagLists consists of several TagList.
+type TagLists []TagList
+
+// SplitTagLists splits a slice of strings into a slice of TagLists using
+// SplitTagList.
+func SplitTagLists(s []string) (l TagLists) {
+	l = make([]TagList, 0, len(s))
+	for _, t := range s {
+		l = append(l, SplitTagList(t))
+	}
+	return l
+}
+
+func (l TagLists) String() string {
+	return fmt.Sprintf("%v", []TagList(l))
+}
+
+// Set updates the TagList's value.
+func (l *TagLists) Set(s string) error {
+	*l = append(*l, SplitTagList(s))
```

Comment:
Because that'd overwrite the previous value.

Let's say we have a flag with a taglist, like this:
```go
var tags restic.TagLists
pflag.Var(&tags, "tags", "use these tags")
```

When the binary is called like this:
```
$ foo --tags foo --tags bar --tags baz,boo,buu
```

then the method `Set()` is called three times:
```go
Set("foo") -> *l = [ ["foo" ]
Set("bar") -> *l = [ ["foo"], ["bar"] ]
Set("baz,boo,buu") -> *l = [ ["foo"], ["bar"], ["baz", "boo", "buu"] ]
```